pom.xml

```diff
@@ -68,23 +68,45 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
+          <configuration>
+            <release>${javaVersion}</release>
```

Comment:
This change caused all Maven modules and test classes to be build using Java 1.6, which is why I removed `@SafeVarargs` from two test classes and adjusted `ProtoTypeAdapter` because `Map.putIfAbsent` was added in Java 8 while `ConcurrentMap.putIfAbsent` already existed in Java 1.6.